### PR TITLE
Feature hiera merge for facts::facts_hash

### PR DESCRIPTION
**Pull request by philippvh**
Adding (optional) hiera_merge feature for facts::facts_hash.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,6 +96,12 @@ Hash of facts to be passed to facter::fact with create_resources().
 
 - *Default*: undef
 
+facts_hash_hiera_merge
+----------------------
+Boolean to control merges of all found instances of facter::facts_hash in Hiera. This is useful for specifying facts entries at different levels of the hierarchy and having them all included in the catalog.
+
+- *Default*: false
```
Comment (ghoneycutt) [inline README.md]: Wouldn't we want to default this to `true` ?

**Comment by ghoneycutt** [comment]
@Phil-Friderici what do you think?

**Comment by philippvh** [comment]
I left the default with "false", because this way the module does not change behavior by my PR. E.g. currently you can make sure on FQDN-level, to set _exactly_ some facts and by that delete other set elsewhere. With merge, this does not work any more and could have impact on existing setups.
I intend to set "true" in our common.yaml, yes.

File: README.md (hunk 1 above)
Comment (philippvh) [inline README.md]: Backward compatibility; I did not want to change the behavior of the module by this PR.

Comment (philippvh) [inline README.md]: I do intend to set it to true in our common.yaml.

**Comment by Phil-Friderici** [comment]
As both of you want to have the functionality as the new default, it seems it is time for a new major version.

**Comment by ghoneycutt** [comment]
@Phil-Friderici agreed. Anything else you might want to change about the module?

**Comment by philippvh** [comment]
@ghoneycutt - maybe issue 41? (Very tiny thing, I did not prepare a PR for this yet.)

**Comment by ghoneycutt** [comment]
Puppet Labs wanted me to change how the facts.txt was handled for the Approved status. I did it there way, but the outcome is that the file can be empty, which causes facter to emit a warning message which is annoying. That should be fixed

**Comment by ghoneycutt** [comment]
Released in v3.1.0

Thanks @Phil-Friderici @philippvh

**Pull request merged**